CalibCalorimetry/EcalTrivialCondModules/src/EcalLaserCondTools.cc

```diff
@@ -0,0 +1,489 @@
+    f_ = fopen(fnames_[ifile_].c_str(), "r");
+    iline_ = 0;
+    if (f_ == 0) {
+      std::cerr << "Failed to open file " << fnames_[ifile_] << ". File skipped!\n";
+    } else {
+      return true;
+    }
+  }
+}
+
+bool EcalLaserCondTools::FileReader::readTime(int& t1, int t2[92], int& t3) {
+  trim();
+  if ((f_ == 0 || feof(f_)) && !nextFile()) {
+    if (verb_ > 1)
+      cout << "No more record\n";
+    return false;
+  }
+  int i;
+  char* buf = 0;
+  size_t s = 0;
+  while ((i = fgetc(f_)) != 'T' && i != 'L' && i >= 0)
+    getline(&buf, &s, f_);
+  if (buf)
+    free(buf);
+  buf = 0;
+
+  if (i == 'L') {  //last record put 3 consecutive times starting from end of prev. IOV
+    t1 = t3;
+    for (int i = 0; i < 92; ++i)
+      t2[i] = t1 + 1;
+    t3 = t1 + 2;
+    return true;
+  }
+
+  if (i != 'T') {
+    if (verb_ > 1)
+      cout << "No more record or bad line type/marker (getc returned " << i << ")\n";
+    return false;
+  }
+
+  EcalLaserAPDPNRatios::EcalLaserTimeStamp t;
+  int n = fscanf(f_, "%d %d", &t1, &t3);
+  for (int i = 0; i < 92; ++i) {
+    int nn = fscanf(f_, "%d", &t2[i]);
+    if (nn != 1)
+      break;
+    n += nn;
+  }
+
+  int nnn = fscanf(f_, " ");
+
+  if (n != (2 + 92) || nnn != 0)
+    throw cms::Exception("LasCorFile") << "File " << fnames_[ifile_] << " line " << iline_
+                                       << ": syntax error. Expecting 'T' marker followed by 94 values: "
+                                          "t1 t2 t3(lme 1) t3(lme 2) ... t3(lme 92)\n";
+
+  return true;
+}
+
+bool EcalLaserCondTools::FileReader::readPs(DetId& detid, EcalLaserAPDPNRatios::EcalLaserAPDPNpair& corr) {
+  if (f_ == 0) {
+    if (verb_)
+      cout << "Requested to read p1..p3 parameter line while no file is closed.\n";
+    return false;
+  }
+
+  trim();
+  int i = fgetc(f_);
+
+  if (i != 'P') {
+    if (verb_ && i >= 0)
+      cout << "File " << fnames_[ifile_] << " line " << iline_ << ": unexpected line type, '" << (char)i
+           << "' while expecting 'P'\n";
+    if (verb_ && i < 0)
+      cout << "Failed to read p1..p3 parameter line\n";
+    return false;
+  }
+
+  int rawdetid;
+  int n = fscanf(f_, "%d %f %f %f\n", &rawdetid, &corr.p1, &corr.p2, &corr.p3);
+  ++iline_;
+
+  if (n != 4) {
+    //    corr.p2=corr.p1;
+    //    corr.p3=corr.p1;
+    throw cms::Exception("I/O") << "Syntax error at line " << iline_ << "of file " << fnames_[ifile_] << " read " << n
+                                << " values,"
+                                << " raw id" << rawdetid << ": " << corr.p1 << ", " << corr.p2;
+  }
+  detid = rawdetid;
+  const int ECALID = 3;
+  if (detid.det() != ECALID)
+    throw cms::Exception("InvalidValue") << "Line " << iline_ << "of file " << fnames_[ifile_]
+                                         << " contains an invalid det ID (detector code is not ECAL!)\n";
+  if (detid.subdetId() == EcalBarrel)
+    EBDetId(detid);  //in order to throw an exception if detid is not valid
```

Comment:
@grasph - your clang warnings are from this line, which doesn't seem like valid code.. What do you actually want to do here?

Reply:
Thanks @davidlange6 to have look at this.

The EBDetId(int) ctor used to do some validity check on the detid and throw an exception in case it is invalid. The purpose of this line was to check we can build an EBDetId from detid. 

Anyway, looking at the code of EBDetId, the check was removed, I guess for performance reason, and therefore the two lines have no more effect. I will remove them.